File: Solutions/Complement_of_Base_10_Integer/Complement_of_Base_10_Integer.cpp
class Solution
{

public:
    int bitwiseComplement(int n)
    {

        if (n == 0)
            return 1;

        int power = 1;

        int ans = 0;

        while (n > 0)
        {
            int bit = n & 1;

            if (bit == 0)
            {
                ans += (1 * power);
            }

            n = n >> 1;

            power = power * 2;
        }

        return ans;
    }
};

//Another
class Solution
{
public:
    int bitwiseComplement(int N)
    {
        //         if(N==0)
        //             return 1;
        //        long int sum=0,i=0;
        //      while(N!=1)
        //      {
        //          if(N%2==0)
        //             sum=sum+pow(2,i)*('1'-'0');
        //          else
        //             sum=sum+pow(2,i)*('0'-'0');
        //         N=N/2;
        //          i++;
        //      }

        //              return sum;
        int c = 1;
        while (c < N)
            c = (c << 1) + 1;
        return N ^ c;
    }
};

File: Solutions/Complement_of_Base_10_Integer/Complement_of_Base_10_Integer.go
func bitwiseComplement(N int) int {
	if N == 0 {
		return 1
	}
	
	NP := N // a copy of N
	for offset := uint(1); N>>offset > 0; offset <<= 1 {
		N |= (N >> offset)
	}
	return NP ^ N
}

//Another
func bitwiseComplement(N int) int {
    if N == 0 {
        return 1
    }
    p2 := 1
    r := 0
    for N > 0 {
        x := N & 1
        if x == 0 {
            r |= p2
        }
        N >>= 1
        p2 <<= 1
    }
    return r
}